src/PrestaShopBundle/EventListener/Admin/Context/ShopContextListener.php

```diff
@@ -183,4 +198,30 @@ private function redirectShopContext(RequestEvent $requestEvent): ?RedirectRespo
             ['setShopContext', 'conf']
         ));
     }
+
+    /**
+     * @throws ReflectionException
+     */
+    private function verifyAllShopAttribute(Request $request): ?ShopConstraint
+    {
+        try {
+            $routeInfo = $this->router->matchRequest($request);
+            $controller = $routeInfo['_controller'];
+            $controllerParts = explode('::', $controller);
+            [$className, $methodName] = $controllerParts;
```

Comment:
I'm nitpicking but I think you don't need the intermediary step here:

```suggestion
            [$className, $methodName] = explode('::', $controller);
```